user-app/src/App.tsx

```diff
@@ -6,24 +6,18 @@ import {
 } from './containers';
 
 import client from './services/architect';
-import { ArchitectAuthProviders } from 'architect-sdk/lib/core/auth';
+import { ArchitectAuthProviders, ArchitectCredentials } from 'architect-sdk/lib/core/auth';
 
 const App = () : React.ReactElement => {
 	const [isAuthenthicated, setIsAuthenthicated] = useState<boolean>(client.isAuthenticated());
 
 	const handleLogin = (
-		email : string,
-		password : string,
+		credentials : ArchitectCredentials,
 		provider : ArchitectAuthProviders,
-		event : React.FormEvent<HTMLFormElement>,
 		setLoading: (value : boolean) => void,
 	) => {
-		event.preventDefault();
 		setLoading(true);
-		client.login({
-			email,
-			password,
-		}, provider)
+		client.login(credentials, provider)
 			.then(() => {
 				setLoading(false);
 				setIsAuthenthicated(client.isAuthenticated());
```

Comment:
problem ako dodje do greske, ovaj then na liniji 27 nece se izvrsiti i klijent nece moci da posalje request (zato sto je loading true).

Reply:
Aka dodje do greske catch na liniji 31 ce se izvrsiti i staviti loading kao false. Catch u ovom slucaju radi zato sto je na samom serveru greska 500 i kod nas vraca CORS sto pokrece ovu gresku. Posto nema konkretnog objekta ili poruke koji ukazuje na gresku na serveru nego se vraca 500 i u browser-u CORS to samim tim pokrece catch sto zaustavlja loading.

Reply:
Sada vidim, ok je.